src/dev/src/app/app.component.ts

```diff
@@ -21,13 +22,15 @@ export class AppComponent {
 
   constructor(
     @Inject(DOCUMENT) private document: Document,
-    @Inject(PLATFORM_ID) private platformId: Record<string, any>
+    @Inject(PLATFORM_ID) private platformId: Record<string, any>,
+    private routeHistoryService: RouteHistoryService
   ) {
     if (isPlatformBrowser(this.platformId)) {
       this.linkRef = this.document.createElement('link');
       this.linkRef.rel = 'stylesheet';
       this.linkRef.href = this.themes[0].href;
       this.document.querySelector('head').appendChild(this.linkRef);
+      this.routeHistoryService.toString(); // needed for angular to create service instance
```

Comment:
this looks very weird... pls either copy the code from the service to app.component.ts or don't do it in constructor and call an "init" method or anything, as constructors shouldn't have any logic inside (according to best practices)

Reply:
what about AppComponent's constructor - seems to have some logic inside ;-)

Reply:
Changed in latest commit

Reply:
one doesn't know best practices when he starts something new :-)